src/test/java/com/intel/gkl/smithwaterman/SmithWatermanUnitTest.java

```diff
@@ -70,6 +71,68 @@ public void inputDataTest() {
         }
     }
 
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void smithWatermanThrowsIllegalArgumentExceptionIfRefSequenceLengthTooLong(){
+        final IntelSmithWaterman sw = new IntelSmithWaterman();
+        int sequenceLength = TestingUtils.MAX_SW_SEQUENCE_LENGTH+1;
+
+        byte[] ref = TestingUtils.generateRandomDNAArray(sequenceLength);
+        byte[] align = new byte[]{'T', 'C', 'C', 'G'};
+
+        SWParameters SWparameters = new SWParameters(10, -5, -10, -10);
+        sw.align(ref, align, SWparameters, SWOverhangStrategy.IGNORE);
+
+        Assert.fail();
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void smithWatermanThrowsIllegalArgumentExceptionIfAlignSequenceLengthTooLong(){
+        final IntelSmithWaterman sw = new IntelSmithWaterman();
+        int sequenceLength = TestingUtils.MAX_SW_SEQUENCE_LENGTH+1;
+
+        byte[] ref = new byte[]{'T', 'C', 'C', 'G'};
+        byte[] align = TestingUtils.generateRandomDNAArray(sequenceLength);
+
+        SWParameters SWparameters = new SWParameters(10, -5, -10, -10);
+        sw.align(ref, align, SWparameters, SWOverhangStrategy.IGNORE);
+
+        Assert.fail();
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void smithWatermanThrowsIllegalArgumentExceptionIfMatchValueGreaterThanMaxAllowed(){
+        final IntelSmithWaterman sw = new IntelSmithWaterman();
+        int matchValue = TestingUtils.MAX_SW_MATCH_VALUE + 1;
+        byte[] ref = new byte[]{'A', 'C', 'C', 'G'};
+        byte[] align = new byte[]{'T', 'C', 'C', 'G'};
+
+        SWParameters SWparameters = new SWParameters(matchValue, -5, -10, -10);
+        sw.align(ref, align, SWparameters, SWOverhangStrategy.IGNORE);
+
+        Assert.fail();
+    }
+
+    @Test(enabled = true)
+    public void maxSequenceFullAlignmentTest(){
+        final IntelSmithWaterman smithWaterman = new IntelSmithWaterman();
+        final boolean isLoaded = smithWaterman.load(null);
+        
+        if(!isLoaded) throw new SkipException("Could not load IntelSmithWaterman; skipping test...");
+
+        int matchValue = TestingUtils.MAX_SW_MATCH_VALUE;
+        int sequenceLength = TestingUtils.MAX_SW_SEQUENCE_LENGTH;
+
+        String expectedCigar = String.format("%dM", sequenceLength);
+
+        byte[] ref = TestingUtils.generateRandomDNAArray(sequenceLength);
+        byte[] align = Arrays.copyOf(ref, ref.length);
+
+        SWParameters SWparameters = new SWParameters(matchValue, -5, -10, -10);
+        SWNativeAlignerResult result = smithWaterman.align(ref, align, SWparameters, SWOverhangStrategy.IGNORE);
```

Comment:
Do we want to consider other SW Overhanf strategies?

Reply:
Overhanging strategies won't make a difference in this case. These strategies determine the way misaligned sequence ends are handled. In this test, we expect a full alignment. For instance, in the `SOFTCLIP` strategy there would be nothing to truncate.